### DISCUSSION TITLE
Post-processing ".out" and ".csv" TOUGH2 output

### DISCUSSION THREAD

**Discussion by medhay36** [acroucher/PyTOUGH#43] [Q&A] [answered]
Hello 

I am actually using TOUGH2 installed on a HPC cluster.

The outputs i get are mainly two files: ".out" file and ".csv" file.

My question is how to use the t2listing package to read one of those files ? or what option to specify in the INFILE of TOUGH2 to obtain a listing file ?

I wonder also if there are other option to obtain a VTK or VTU file in order to prepare isosurfaces and more developed graphs not only time series of pressure and temperature ?

and thank you very much for your help

**Comment by acroucher** [comment, marked as answer]
The *.out file should be a listing file which you can read using `t2listing`.

Once you have read your results into a `t2listing` object you can use its `write_vtk()` method to produce *.vtu output for visualisation. To do that, you will also need to have a `mulgrid` object representing your grid geometry. If you don't have one of those, but your grid is rectangular, you might be able to reverse-engineer a `mulgrid` geometry using the `t2grid` `rectgeo()` method.

**Reply by medhay36** [reply]
![image](https://user-images.githubusercontent.com/122643918/212956745-fc2b5cba-d9d1-4b11-b6be-427b220ce5f4.png)

thank you for your answer professor 

actually i tried to read my results (the .out file) into a t2lisitng object but i encouter errors.

pytough libraries seems to be loaded correctly and i used them to generate the infile without problems. 

have some of pythough users encoutred similar errors previously in the post-processing ?

**Reply by acroucher** [reply]
It looks like you have not imported the `t2listing` module. What happens if you add `from t2listing import *` to your imports?

**Reply by medhay36** [reply]
no errors dear professor
the module seems to be loaded correctly 

![image](https://user-images.githubusercontent.com/122643918/213028053-e82ff91c-c375-451a-b902-bc060ed2abea.png)

**Reply by medhay36** [reply]
> It looks like you have not imported the `t2listing` module. What happens if you add `from t2listing import *` to your imports?

Hello dear professor @acroucher , 

I've been working since a while on this problem.

Unfortunately, i couldn't solve the problem.

I encounter the following error despite i have the original geometry file .geo and prepare the mulgrid from it 
![image](https://user-images.githubusercontent.com/122643918/234143623-ade6431b-6cd6-4265-a75b-0c5b18af1089.png)

**Reply by acroucher** [reply]
OK, so this looks like a different error now. I can't tell what might be causing it, without seeing the listing file (INFILE.out). Are you able to email it to me? You can find my email address [here](https://profiles.auckland.ac.nz/a-croucher).
